source/common/http/http1/codec_impl.cc

```diff
@@ -459,6 +459,7 @@ ConnectionImpl::ConnectionImpl(Network::Connection& connection, CodecStats& stat
       max_headers_kb_(max_headers_kb), max_headers_count_(max_headers_count) {
   output_buffer_.setWatermarks(connection.bufferLimit());
   http_parser_init(&parser_, type);
+  parser_.allow_chunked_length = 1;
```

Comment:
Given the danger of smuggling I'd be inclined to disable this by default, and allow it only iff it's configured on via API or runtime hooks.  
@mattklein123 @antoniovicente for second opinion.

If they rule I'm being overly concerned, we'll instead want to runtime guard as it's a behavioral change, and either way we will want release notes and tests of the new behavior :-)

Reply:
+1. I think this should be a config option that is set on the HTTP protocol options? That way it can be set independently for upstream and downstream? WDYT? It shouldn't be hard to add this as the options are already plumbed through everywhere IIRC.

Reply:
The idea was to:
 * enable non-safe parser configuration by default in envoy
 * enable strict validation by default in envoy (so we don't rely on parser behavior)
and on top of that
* add configuration option (http protocol options) to disable strict validation in envoy.

I can update diff to enable non-safe parser mode only if configuration option is set. If so - should I leave sanity checking enabled by default (even if we know default parser won't allow such request)?

Reply:
I think it's fine to do what you propose (on all the time in parser, block in Envoy code), but will defer to @alyssawilk if she has a strong opinion on it.

Reply:
Yeah I'm fine with that, as long as we double check our tests as we go to make sure we're actually doing the blocking in envoy.